File: MosPosudit.UI/shared/lib/dtos/category/category_update_request.dart
class CategoryUpdateRequestDto {
  final String? name;
  final String? description;
  final String? imageBase64;

  CategoryUpdateRequestDto({
    this.name,
    this.description,
    this.imageBase64,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{};
    if (name != null) json['name'] = name;
    if (description != null) json['description'] = description;
    if (imageBase64 != null) json['imageBase64'] = imageBase64;
    return json;
  }
}

File: MosPosudit.Model/Requests/Category/CategoryUpdateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MosPosudit.Model.Requests.Category
{
    public class CategoryUpdateRequest
    {
        [StringLength(100, ErrorMessage = "Category name cannot be longer than 100 characters")]
        public string? Name { get; set; }

        [StringLength(500, ErrorMessage = "Description cannot be longer than 500 characters")]
        public string? Description { get; set; }

        public string? ImageBase64 { get; set; }
    }
}

